pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -204,25 +204,21 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		return nil, fmt.Errorf("failed to get mounted cgroup subsystems: %v", err)
 	}
 
-	if failSwapOn {
-		// Check whether swap is enabled. The Kubelet does not support running with swap enabled.
-		swapFile := "/proc/swaps"
-		swapData, err := os.ReadFile(swapFile)
-		if err != nil {
-			if os.IsNotExist(err) {
-				klog.InfoS("File does not exist, assuming that swap is disabled", "path", swapFile)
-			} else {
-				return nil, err
-			}
-		} else {
-			swapData = bytes.TrimSpace(swapData) // extra trailing \n
-			swapLines := strings.Split(string(swapData), "\n")
+	swap.InitializeRootDir(nodeConfig.KubeletRootDir)
+	isSwapOn, err := swap.IsSwapOn()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine if swap is on: %w", err)
+	}
 
-			// If there is more than one line (table headers) in /proc/swaps, swap is enabled and we should
-			// error out unless --fail-swap-on is set to false.
-			if len(swapLines) > 1 {
-				return nil, fmt.Errorf("running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false. /proc/swaps contained: %v", swapLines)
-			}
+	if isSwapOn {
+		if failSwapOn {
+			return nil, fmt.Errorf("running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false")
```

Comment:
nit: is the `!` intentional after `swap`?

Reply:
I took it as two sentences being there, with the first one being very assertive :smile:

Reply:
if that's intentional then the nit would be capitalize Or (though I think I prefer them as the same sentence)

Reply:
Yeah. It might have been kept all lower-case since this is an error message.

The entire message could, perhaps, be improved too... Hmm.

Reply:
This message preexisted this PR actually, I've just moved it:
https://github.com/kubernetes/kubernetes/blob/8362f51c6620a55953ac564d24fa82d2f2ea2403/pkg/kubelet/cm/container_manager_linux.go#L224

But yeah, this `!` seems out of place, I'll remove it